### ISSUE TITLE
Whitelist with no cache is not evaluating on every request

### ISSUE THREAD

**Issue by adrianRiobo** [istio/istio#7766]
**Describe the bug**
Configuring an adapter for whitelist with no cache it is supposed to evaluate on every request but after a request is whitelisted subsequent request (random small period of time 10s-30s) are not evaluated and pass as whitelisted even if the do not match the whitelist. 

**Expected behavior**
Every request should be evaluated and if they do not match it should response with 404.

**Steps to reproduce the bug**
1. Install BookingInfo sample from istio.

2. Create an adapter with no cache:

```
apiVersion: config.istio.io/v1alpha2
kind: listchecker
metadata:
  name: apikey-test
spec:
  overrides:["b906014b43ae06db5ff572a13178aaec3260a1b2"]
  #Check on every request
  ttl: 0s
  cachingInterval: 0s
  cachingUseCount: 0
  blacklist: false
```

3. Create a template:

```
apiVersion: config.istio.io/v1alpha2
kind: listentry
metadata:
  name: apikey-test
spec:
  value: request.api_key | ""
```

4. Create a rule:

```
apiVersion: config.istio.io/v1alpha2
kind: rule
metadata:
  name: apikey-test
spec:
  match: destination.labels["app"] == "productpage"
  actions:
  - handler: apikey-test.listchecker
    instances:
    - apikey-test.listentry

```
5. Send a request with no x-api-key header to http://$GATEWAY_URL:31280/productpage 
    Get a 404 as x-api-key is not whitelisted

6.  Send a request with x-api-key=b906014b43ae06db5ff572a13178aaec3260a1b2 to http://$GATEWAY_URL:31280/productpage 
     Get a 200 an see the page as x-api-key is whitelisted

7. Send in a short period of time a request with no x-api-key to http://$GATEWAY_URL:31280/productpage
     Get a 200 where x-api-key has no value so it is not whitelisted 

**Version**

Istio
Version: 1.0.0
GitRevision: 3a136c90ec5e308f236e0d7ebb5c4c5e405217f4
User: root@71a9470ea93c
Hub: gcr.io/istio-release
GolangVersion: go1.10.1
BuildStatus: Clean

Kubernetes
Kubernetes v1.9.5

**Is Istio Auth enabled or not?**
Installed stable isitio.yaml (no auth)

**Environment**
On premise k8s cluster in CentOS Linux release 7.2.1511 (Core)

**Cluster state**
Attached file
[istio-dump.tar.gz](https://github.com/istio/istio/files/2273317/istio-dump.tar.gz)

**Comment by adrianRiobo** [comment]
Issue closed. This problem is solved whit new istio version: 1.0.2

**Comment by dotw** [comment]
this is not fixed in 1.0.2.
the whitelist is indeed affecting the request when it is first applied, but with several requests, it will be not working anymore.  Waiting for a period, it works again, and then not works,...
